lebedigital/calibration/calibrationWorkflow.py

```diff
@@ -0,0 +1,202 @@
+        sensor_values={
+            'nu': 0.2,
+            'height': exp_output['height'],
+            'radius': exp_output['diameter'] / 2,
+            'displacement_list': exp_output['displacement'],
+            linear_elasticity.output_sensor.name: y_test,
+        },
+    )
+
+    # ==============================================
+    #       Add likelihood model(s)
+    # ==============================================
+
+    # add the likelihood model to the problem
+    problem.add_likelihood_model(
+        GaussianLikelihoodModel(
+            prms_def="sigma",
+            experiment_name=experiment_name,
+            model_error="additive",
+            name="SimpleLikelihoodModel",
+        )
+    )
+
+    # give problem overview
+    problem.info()
+
+    # ==============================================
+    #        Export knowledge graph
+    # ==============================================
+
+    # create the knowledge graph and print it to file
+    #dir_path = os.path.dirname(__file__)
+    dir_path = calibrated_data_path
+    basename_owl = os.path.basename(__file__).split(".")[0] + ".owl"
+    knowledge_graph_file = os.path.join(dir_path, basename_owl)
+    export_knowledge_graph(problem, knowledge_graph_file, data_dir=dir_path)
+
+    # ========================================================================
+    #        Solve problem with inference engine and write results to graph
+    # ========================================================================
+
+    # run inference step using emcee
+    emcee_solver = EmceeSolver(problem, show_progress=True)
+
+    if mode == 'cheap':
+        inference_data = emcee_solver.run_mcmc(
+            n_walkers=4,
+            n_steps=1,
+            n_initial_steps=4,
+        )
+    else:
+        inference_data = emcee_solver.run_mcmc(
+            n_walkers=5,
+            n_steps=40,
+            n_initial_steps=5,
+        )
+
+    # export the results from the 'inference_data' object to the graph
+    export_results_to_knowledge_graph(
+        problem,
+        inference_data,
+        knowledge_graph_file,
+        data_dir=dir_path,
+    )
+
+    # =======================================================================
+    #                          Query the knowledge graph
+    # =======================================================================
+
+    # get the samples from the knowledge graph
+    sample_dict = import_parameter_samples(knowledge_graph_file)
+
+    # ========================================================================
+    #       Posterior Predictive
+    # ========================================================================
+    nu = 0.2
+    E_pos = sample_dict['E'] * 1000  # N/mm2 ~ E_mean ~ 95E03N/mm2
+    # three_point = three_point_bending_example()
+    pos_pred = PosteriorPredictive(three_point_bending_example, known_input_solver=nu, parameter=E_pos)
+
+    if mode == 'cheap':
```

Comment:
what is the return value of this method?

Reply:
Is there a test for this functionality? (so where we know what we would expect?)

Reply:
@joergfunger: which method?

Reply:
for the inference method. Usually, all the methods in the lebedigital modul should be part of the pytest set, such that errors can be detected.

Reply:
For teh calibration script there is a test. If you mean for teh posterior predictive here, there is none and can be added